ccmlib/utils/version.py

```diff
@@ -28,16 +28,16 @@
 class ComparableScyllaVersion:
     """Accepts and compares known 'non-semver' and 'semver'-like Scylla versions."""
 
-    def __init__(self, version_string: str):
+    def __init__(self, version_string: str) -> None:
         parsed_version = self.parse(version_string)
-        self.v_major = int(parsed_version[0])
-        self.v_minor = int(parsed_version[1])
-        self.v_patch = int(parsed_version[2])
-        self.v_pre_release = parsed_version[3] or ''
-        self.v_build = parsed_version[4] or ''
```

Comment:
In addition to the commit message, this had a bug if "prerelease" is missing and "build" exists, then it would have been in position 3, not 4.

Reply:
The original code has a set of regex and "fix" that align the version to a strict semver format, so probably so of the assumptions here counted on that.